store/copr/batch_coprocessor.go

```diff
@@ -616,6 +622,31 @@ func buildBatchCopTasks(bo *backoff.Backoffer, store *kvStore, ranges *KeyRanges
 				zap.Int("task len", len(batchTasks)))
 		}
 		metrics.TxnRegionsNumHistogramWithBatchCoprocessor.Observe(float64(len(batchTasks)))
+
+		// generate tableRegions for batchCopTasks
+		if partitionIDs != nil {
+			for _, copTask := range batchTasks {
+				tableRegions := make([]*coprocessor.TableRegions, len(partitionIDs))
+				// init coprocessor.TableRegions
+				for j, pid := range partitionIDs {
+					tableRegions[j] = &coprocessor.TableRegions{
+						PhysicalTableId: pid,
+					}
+				}
+				// fill region infos
+				for _, ri := range copTask.regionInfos {
+					tableRegions[ri.PartitionIndex].Regions = append(tableRegions[ri.PartitionIndex].Regions,
+						ri.toCoprocessorRegionInfo())
+				}
+				// clear empty table region
+				for j := len(tableRegions) - 1; j >= 0; j-- {
+					if len(tableRegions[j].Regions) == 0 {
+						tableRegions = append(tableRegions[:j], tableRegions[j+1:]...)
+					}
+				}
+				copTask.PartitionTableRegions = tableRegions
```

Comment:
why not set copTask.regionInfos to nil here?

Reply:
addressed.